api/src/app/modules/attachment/attachment.service.ts

```diff
@@ -164,8 +164,9 @@ export class AttachmentService extends DataService<Attachment, Repository<Attach
     // NestJS/TypeORM has breaking change for find API which only accespts 'option' argument and
     // no 'id' for findOne(id, options) like before. We need to specifically build the 'options.where'
     // for the TypeORM api now.
-    const revisedOptions = this.addCommonRelationsToFindOptions(this.addCommonRelationsToFindOptions(
-        { select: [ 'id', 'projectId', 'fileName', 'attachmentType' ] }));
+    const revisedOptions = this.addCommonRelationsToFindOptions(
+        { select: [ 'id', 'projectId', 'fileName', 'attachmentType']}
```

Comment:
This logic is still weird. I think we should just be calling this.findEntityWithCommonRelations(id) instead of all of this logic (up to line 171) which is duplicating (incorrectly) the generic logic in data.service. The existing logic is specifying what columns to return, so that's why attachmentTypeCode is null as it isn't included in the list. 

I'm trying to assess if there's a reason we did the logic special this way, but I can't think of one at the moment.

Reply:
Ah~, I probably understand now with the previous code. It seems it is trying to **reduce** "**select**" return from repository.
So, other functions like "**getFileContent**" that uses "**this.findFileDatabase(id, user)**" function only needs these fields' value from **Attachment Entity**:

> **id,
> projectId,
> fileName**

plus, it also needs "**attachmentType**"(entity) or "**attachmentTypeCode**"(just code) (either one) to do validation on `this.isViewAuthorized`.
See the calling function:
```
async getFileContent(id: number, user?: User): Promise<AttachmentFileResponse> {
    const attachmentFileResponse = await this.findFileDatabase(id, user);

    //Creating the objectName for the Object Storage
    const objectName = this.createObjectUrl(attachmentFileResponse.projectId, attachmentFileResponse.id, attachmentFileResponse.fileName )
```
That's probably why it needs a list of **["id", "projectId", "fileName", "attachmentType"]**

Reply:
But on the other hand, the attachment.service.ts has 
```  
protected getCommonRelations(): string[] {
    return ['attachmentType'];
  }
```
So basically, they are equivalent. Although "**project**" entity isn't listed as common relation in both of them( through `this.findEntityWithCommonRelations(id)` or through specific selection `["id", "projectId", "fileName", "attachmentType"]`), "**project**" entity is always returned (as "eager" fetch I guess).
I will change to use `this.findEntityWithCommonRelations(id)`.